distribution/docker/Dockerfile

```diff
@@ -18,6 +18,7 @@
 #
 
 ARG JDK_VERSION=11
+ARG BASE_IMAGE=gcr.io/distroless/java$JDK_VERSION-debian11
```

Comment:
it's usually preferable to define docker arguments as late as possible to maximize the number of layers that can be cached and reused.

Defining BASE_IMAGE at this level, means the entire builder step cache will be invalidated if the base image is changed, when that is actually unnecessary